sql/sql_zip_dict.cc

```diff
@@ -103,13 +104,27 @@ bool bootstrap(THD *thd) {
   DBUG_EXECUTE_IF("skip_compression_dict_create", skip_bootstrap = true;
                   return false;);
 
+  dd::String_type dict_table_str_enc{dict_table_str};
+
+  if (dd::bootstrap::DD_bootstrap_ctx::instance()
+          .is_system_tablespace_encrypted()) {
+    dict_table_str_enc += " ENCRYPTION='Y'";
```

Comment:
Why is this required? I mean which tables need this?

Reply:
Ok, nevermind, found the definition above. But I don't think it is right strategy.

We shouldn't add "ENCRYPTION=Y" to each create table statement. 

can we set mysql database default encryption as encrypted instead of fixing each "create table mysql. *" ?

Reply:
I guess we can change DEFAULT ENCRYPTION on start of a bootstrap and then change it back to the original DEFAULT ENCRYPTION. However, check mysql_system_tables.sql - they are assigning ENCRYPTION per clause - so only advantage we will get are those two tables - I do not think it is worth the effort. Note that mysql_system_tables_fix.sql you are referring to later is about upgrade and for this we need explicit ENCRYPTION clause - DEFAULT ENCRYPTION will not help.

Reply:
Why are we checking "system tablespace" encryption status? Is this the ibdata1 status or mysql.ibd status?

Reply:
change the name of the variable to is_dd_encrypted. Now DD does not see InnoDB internals. Whether to encrypt or not is based on encryption flag from first tablespace in bootstrap.cc:DDSE_dict_init. First Plugin_tablespace is DD tablespace.

Reply:
this variable should be named is_mysql_tablespace_encrypted in the first place - now is is_dd_encrypted